### PR TITLE
SW-27107 - Add reusable workflow for plugins

### PR DESCRIPTION
**Pull request by rathmerdominik**

### CODE REVIEW

File: .github/workflows/setup-shopware.yml [hunk 1]
```diff
@@ -1,10 +1,7 @@
 on:
     workflow_call:
     push:
-#    pull_request:
```
[the remark this answers is not in the document]
Comment (rathmerdominik) [reply .github/workflows/setup-shopware.yml:4]: @PascalThesing This is removable right?

[the remark this answers is not in the document]
Comment (PascalThesing) [reply .github/workflows/setup-shopware.yml:4]: No, this is therefore, that the container are only rebuild if the merge was done into the main branch

**Comment by mitelg** [comment]
can we change the building of the docker images for the older SW versions to manual? because there won't happen any changes anymore in SW 5.0 - 5.6

File: .github/workflows/setup-shopware.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mitelg) [reply .github/workflows/setup-shopware.yml:4]: according to @PascalThesing this code must not be commented. is this correct Pascal?

**Comment by rathmerdominik** [comment]
> can we change the building of the docker images for the older SW versions to manual? because there won't happen any changes anymore in SW 5.0 - 5.6

I added a reusable workflow for this
Sadly, we can only test if this works when it is merged...

File: .github/workflows/setup-current-shopware.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                        php: '8.0'
+                        mysql: '8.0'
+                        consentManager: 'none'
+
+                    -   shopware: '5.7.4'
+                        php: '8.1'
+                        mysql: '8.0'
+                        consentManager: 'none'
+
+                    -   shopware: '5.7.4'
```
Comment (mitelg) [inline .github/workflows/setup-current-shopware.yml:47]: only PHP 8.2 should be sufficient

**Review by mitelg — APPROVED** [review]

**Pull request merged**